src/core/sock.c

```diff
@@ -152,6 +152,12 @@ int nn_sock_init (struct nn_sock *self, struct nn_socktype *socktype, int fd)
     /*  Should be pretty much enough space for just the number  */
     sprintf(self->socket_name, "%d", fd);
 
+    /* Security attribute*/
+    self->sec_attr = NULL;
+    self->sec_attr_size = 0;
+    self->inbuffersz = 4096;
+    self->outbuffersz = 4096;
```

Comment:
That's an ultra-low barrier first step. Wrapped in `#IFNDEF` is a quick way to make it "tunable" externally if you wanted. Anything further, we could keep as a `nice to have` until someone contributes.